### PR TITLE
Recognize single and double quoted arguments.

### PR DESCRIPTION
**Pull request by pavel-sakun**
@mattyb149

### CODE REVIEW

**Comment by pavel-sakun** [comment]
@mattyb149

File: src/main/java/org/pentaho/di/job/entries/spark/JobEntrySparkSubmit.java [hunk 1]
```diff
@@ -542,6 +544,51 @@ public void check( List<CheckResultInterface> remarks, JobMeta jobMeta, Variable
...
+   *
+   * @param commandLineString
+   *          A command line string.
+   * @param variableSpace
+   *          Context for resolving variable names. If {@code null}, no variable resolution will happen.
+   * @return List of parsed arguments
+   * @throws IOException
+   *           when the command line could not be parsed
+   */
+  public List<String> parseCommandLine( String commandLineString ) throws IOException {
```
Comment (mattyb149) [inline src/main/java/org/pentaho/di/job/entries/spark/JobEntrySparkSubmit.java]: Do we not have a third-party library or existing PDI method to parse a command line?

Comment (pavel-sakun) [inline src/main/java/org/pentaho/di/job/entries/spark/JobEntrySparkSubmit.java]: Do not remember being aware of any in PDI, used the code from Sqoop job entry. Could you point the one in PDI? Ant's Commandline might be used as an option.

**Comment by pavel-sakun** [comment]
@mattyb149 @mbradbury2014

File: src/main/java/org/pentaho/di/job/entries/spark/JobEntrySparkSubmit.java (hunk 1 above)
Comment (mattyb149) [inline src/main/java/org/pentaho/di/job/entries/spark/JobEntrySparkSubmit.java]: I looked and couldn't find one, my mistake (although we really should have one).  Will the tokenizer handle keyboards in other languages?

File: src/main/java/org/pentaho/di/job/entries/spark/JobEntrySparkSubmit.java [hunk 2]
```diff
@@ -572,6 +574,49 @@ public void check( List<CheckResultInterface> remarks, JobMeta jobMeta, Variable
...
+    List<String> args = new ArrayList<String>();
+    StringReader reader = new StringReader( commandLineString );
+    try {
+      StreamTokenizer tokenizer = new StreamTokenizer( reader );
+      // Treat a dash as an ordinary character so it gets included in the token
+      tokenizer.ordinaryChar( '-' );
+      tokenizer.ordinaryChar( '.' );
+      tokenizer.ordinaryChars( '0', '9' );
+      // Treat all characters as word characters so nothing is parsed out
+      tokenizer.wordChars( '\u0000', '\uFFFF' );
```
Comment (pavel-sakun) [inline src/main/java/org/pentaho/di/job/entries/spark/JobEntrySparkSubmit.java]: It takes care of 0000-FFFF which seems sufficient.

**Pull request merged**